In [1]:
#import data manipulation library
import numpy as np
import pandas as pd
#import data visualization library
import matplotlib.pyplot as plt
import seaborn as sb
#import filter warning libraries
#Filterwarning will hide all warning messages
import warnings
warnings.filterwarnings('ignore')
#import data logging
import logging
logging.basicConfig(level=logging.INFO,
                    filemode='w',
                    filename='app.log',
                    format='%(asctime)s - %(levelname)s - %(message)s')

In [2]:
logging.info('Please Upload Dataset For EDA...')

In [3]:
data=pd.read_csv('bank-full.csv',sep=';')

In [4]:
logging.info('The Dataset is Successfully Uploaded')

In [5]:
#Import Data Sample
data.sample(frac=1)

age            job  marital  education default  balance housing loan  \
1056    44    blue-collar  married    primary      no     -171     yes   no   
5327    28     technician   single   tertiary      no      370     yes   no   
26424   31    blue-collar  married  secondary      no      268     yes   no   
13226   38      housemaid  married  secondary      no     1508      no  yes   
44576   24        student   single   tertiary      no     1592      no   no   
...    ...            ...      ...        ...     ...      ...     ...  ...   
16534   35     management   single   tertiary      no      657     yes   no   
39361   20    blue-collar   single  secondary      no       76     yes   no   
28240   48       services  married    primary      no     1496     yes   no   
19230   30  self-employed   single   tertiary      no     1495      no   no   
36877   45    blue-collar  married  secondary      no     3092     yes  yes   

         contact  day month  duration  campaign  pdays  previous poutcome    y  
1056     unknown    7   may       441         2     -1         0  unknown   no  
5327     unknown   23   may       108         5     -1         0  unknown   no  
26424   cellular   20   nov        93         2     -1         0  unknown   no  
13226   cellular    8   jul       306         2     -1         0  unknown   no  
44576   cellular   16   aug        98         3    111         5    other   no  
...          ...  ...   ...       ...       ...    ...       ...      ...  ...  
16534   cellular   23   jul       547         2     -1         0  unknown   no  
39361   cellular   18   may       639         2     -1         0  unknown  yes  
28240  telephone   29   jan       284         2    258         2  failure   no  
19230   cellular    6   aug       898         2     -1         0  unknown   no  
36877   cellular   12   may       180         2     -1         0  unknown   no  

[45211 rows x 17 columns]

In [8]:
data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        45211 non-null  int64 
 1   job        45211 non-null  object
 2   marital    45211 non-null  object
 3   education  45211 non-null  object
 4   default    45211 non-null  object
 5   balance    45211 non-null  int64 
 6   housing    45211 non-null  object
 7   loan       45211 non-null  object
 8   contact    45211 non-null  object
 9   day        45211 non-null  int64 
 10  month      45211 non-null  object
 11  duration   45211 non-null  int64 
 12  campaign   45211 non-null  int64 
 13  pdays      45211 non-null  int64 
 14  previous   45211 non-null  int64 
 15  poutcome   45211 non-null  object
 16  y          45211 non-null  object
dtypes: int64(7), object(10)
memory usage: 5.9+ MB


## Data Insights:
1.The dataset contains 45211 rows and 17 columns.

2.Seven Numerical Columns and 10 Categorical Columns,Out of which y is the target.
